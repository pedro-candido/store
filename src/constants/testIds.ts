const testIds = {
  addToCartButton: 'add-to-cart-button',
  removeFromCartButton: 'remove-from-cart-button',
  productTestId: 'product-',
  backButton: 'back-button',
  cartButton: 'cart-button',
};

export { testIds };
