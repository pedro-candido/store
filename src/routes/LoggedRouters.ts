export enum LoggedRouters {
  PRODUCTS = 'Products',
  CART = 'Cart',
}

export type LoggedNavigationParams = {
  [LoggedRouters.CART]: undefined;
  [LoggedRouters.PRODUCTS]: undefined;
};
