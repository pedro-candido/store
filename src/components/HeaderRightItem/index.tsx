import { useNavigation } from '@react-navigation/core';
import { StackNavigationProp } from '@react-navigation/stack';

import HeaderItem from '@components/HeaderItem';
import { testIds } from '@constants/testIds';
import { LoggedNavigationParams, LoggedRouters } from '@routes/LoggedRouters';

const { cartButton } = testIds;

function HeaderRightItem() {
  const { navigate } = useNavigation<StackNavigationProp<LoggedNavigationParams>>();

  const handlePress = () => {
    navigate(LoggedRouters.CART);
  };

  return (
    <HeaderItem
      testID={cartButton}
      hasMarginRight={true}
      iconName={'shopping-cart'}
      onPress={handlePress}
    />
  );
}

export default HeaderRightItem;
