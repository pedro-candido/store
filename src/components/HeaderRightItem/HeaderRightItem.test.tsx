import React from 'react';
import { View } from 'react-native';

import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { render, fireEvent, screen } from '@testing-library/react-native';

import { testIds } from '@constants/testIds';
import { LoggedRouters } from '@routes/LoggedRouters';

import HeaderRightItem from './';

const { cartButton } = testIds;
const mockedNavigate = jest.fn();

const Stack = createStackNavigator();

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: mockedNavigate,
  }),
}));

const Component = () => <View>Cart Screen</View>;

const MockNavigator = () => (
  <NavigationContainer>
    <Stack.Navigator initialRouteName={'TestScreen'}>
      <Stack.Screen name="TestScreen" component={HeaderRightItem} />
      <Stack.Screen name={LoggedRouters.CART} component={Component} />
    </Stack.Navigator>
  </NavigationContainer>
);

describe('HeaderRightItem', () => {
  it('navigates to CART when pressed', () => {
    render(<MockNavigator />);

    const button = screen.getByTestId(cartButton); // Adjust test ID as needed
    fireEvent.press(button);

    expect(mockedNavigate).toHaveBeenCalled();
  });
});
