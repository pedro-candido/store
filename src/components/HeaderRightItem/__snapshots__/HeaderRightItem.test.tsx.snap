// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderRightItem should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hasMarginRight={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginRight": 20,
        "marginTop": "5%",
        "transition": "0.4s",
      },
      {},
    ]
  }
  testID="header-right-item-button"
  type="primary"
>
  <Text />
</View>
`;
